File: src/WorkBC.Data/Model/JobBoard/SavedCareerProfile.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace WorkBC.Data.Model.JobBoard
{
    public class SavedCareerProfile
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Column("NocCodeId2021")]
        public int NocCodeId2021 { get; set; }

        [ForeignKey("JobSeeker")]
        public string AspNetUserId { get; set; }

        [JsonIgnore]
        public virtual JobSeeker JobSeeker { get; set; }

        public DateTime DateSaved { get; set; }
        public DateTime? DateDeleted { get; set; }
        public bool IsDeleted { get; set; } = false;
    }
}

File: jobboard-laravel/app/Models/SavedCareerProfile.php
<?php

namespace App\Models;

use App\Models\Concerns\SoftDeletesByFlag;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * ACCT-6 — a job seeker's saved career (NOC) profile.
 *
 * `NocCodeId2021` is the profile identity. The legacy
 * `EDM_CareerProfile_CareerProfileId` column is vestigial: the .NET
 * CareerProfilesController always wrote it as null and never looked anything up
 * by it, so it is mapped but never populated here either.
 */
final class SavedCareerProfile extends Model
{
    use SoftDeletesByFlag;

    protected $table = 'SavedCareerProfiles';

    protected $primaryKey = 'Id';

    protected $keyType = 'int';

    public $incrementing = true;

    public $timestamps = false;

    protected $casts = [
        'IsDeleted' => 'boolean',
        'NocCodeId2021' => 'integer',
        'DateSaved' => 'datetime',
        'DateDeleted' => 'datetime',
    ];

    public function jobSeeker(): BelongsTo
    {
        return $this->belongsTo(JobSeeker::class, 'AspNetUserId', 'Id');
    }

    public function nocCode(): BelongsTo
    {
        return $this->belongsTo(NocCode2021::class, 'NocCodeId2021', 'Id');
    }
}
